src/canvas/quadFeature.js

```diff
@@ -0,0 +1,123 @@
+//////////////////////////////////////////////////////////////////////////////
+/**
+ * Create a new instance of class quadFeature
+ *
+ * @class
+ * @param {Object} arg Options object
+ * @extends geo.quadFeature
+ * @returns {geo.canvas.quadFeature}
+ */
+//////////////////////////////////////////////////////////////////////////////
+geo.canvas.quadFeature = function (arg) {
+  'use strict';
+
+  if (!(this instanceof geo.canvas.quadFeature)) {
+    return new geo.canvas.quadFeature(arg);
+  }
+  geo.quadFeature.call(this, arg);
+
+  var m_this = this,
+      s_exit = this._exit,
+      s_init = this._init,
+      s_update = this._update,
+      m_quads;
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Build this feature
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._build = function () {
+    if (!m_this.position()) {
+      return;
+    }
+    m_quads = this._generateQuads();
+  };
+
+  /**
+   * Render all of the color quads using a single mapper.
+   *
+   * @param renderState: the render state used for the render.
+   */
+  this._renderColorQuads = function (renderState) {
+      // ....
+      // Not implemented yet.
+  };
+
+  /**
+   * Render all of the image quads using a single mapper.
+   *
+   * @param renderState: the render state used for the render.
+   */
+  this._renderImageQuads = function (context2d, map) {
+    if (!m_quads.imgQuads.length) {
+      return;
+    }
+
+    $.each(m_quads.imgQuads, function (idx, quad) {
+      var w = quad.image.width,
```

Comment:
Please add a guard before this of the form
```
if (!quad.image) {
  return;
}
```
of something similar.  When a quad is waiting for the image to load, the image attribute may not be present.